src/System.Numerics.Experimental/Half.cs

```diff
@@ -353,5 +356,402 @@ public bool TryFormat(Span<char> destination, out int charsWritten, ReadOnlySpan
         {
             throw new NotImplementedException();
         }
+
+        // -----------------------Start of to-half conversions-------------------------
+
+        public static implicit operator Half(int i)
+        {
+            bool sign = i < 0;
+            Half h = (uint)(sign ? -i : i); // Math.Abs but doesn't throw exception, because we cast it to uint anyway
+            return sign ? new Half((ushort)(h.m_value | SignMask)) : h;
+        }
+
+        public static implicit operator Half(uint i)
+        {
+            if (i == 0)
```

Comment:
Would be good to keep this as part of the `if (shiftDist >= 0)` case. Having it here will cause a mispredicted branch for all non-zero inputs and will force 2 branch checks for any non-zero result.

Having it as part of the `shiftDist` check will result in 1 or 2 branch checks, depending on the input and can put `0` on the mispredict path.